File: src/types/character.ts
export type CharacterViewStatus = 'queued' | 'generating' | 'completed' | 'failed';

export interface CharacterView {
  id: number;
  character_id: number;
  /** 生成中的占位视图可能为空串 */
  image_url: string | null;
  view_type?: 'front' | 'side' | 'back' | 'expression' | 'action' | null;
  sort_order: number;
  status: CharacterViewStatus;
  error_message?: string | null;
  /** 本次生成是否参考了角色的种子图 */
  use_seed_image: boolean;
  created_at: string;
}

/** 声音档案配置 */
export interface VoiceConfig {
  /** 音频样本文件 URL */
  audio_url?: string;
  /** 音频文件名（展示用） */
  audio_name?: string;
  /** 声音名称/标签（如"温柔女声"） */
  voice_name?: string;
  /** 第三方 TTS 平台的 voice_id（可选） */
  voice_id?: string;
}

export interface Character {
  id: number;
  name: string;
  description?: string;
  visual_prompt?: string;
  seed_image_url?: string;
  voice_config?: VoiceConfig | null;
  views?: CharacterView[];
  creator_id: number;
  created_at: string;
  updated_at: string;
}

export interface CreateCharacterRequest {
  name: string;
  description?: string;
  visual_prompt?: string;
  voice_config?: VoiceConfig;
}

export interface GenerateViewsRequest {
  count: number;
  view_types: string[];
  /** 是否参考角色的种子图（默认 false） */
  use_seed_image?: boolean;
  /** 图像模型 id（AI_IMAGE_MODELS 中某一项 id）；不传走默认模型 */
  model_id?: string;
}

File: backend/app/schemas/character.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class CharacterViewOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    character_id: int
    # 生成占位时 image_url 可能为空
    image_url: Optional[str] = None
    view_type: Optional[str] = None
    sort_order: int
    status: str = "completed"
    error_message: Optional[str] = None
    # 本次生成是否参考了角色种子图
    use_seed_image: bool = False
    created_at: datetime


class CharacterOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: Optional[str] = None
    visual_prompt: Optional[str] = None
    seed_image_url: Optional[str] = None
    voice_config: Optional[dict] = None
    views: list[CharacterViewOut] = Field(default_factory=list)
    creator_id: int
    created_at: datetime
    updated_at: datetime


class GenerateViewsRequest(BaseModel):
    count: int = Field(ge=1, le=20)
    view_types: list[str]
    # 是否参考角色种子图；默认 False，向后兼容老前端
    use_seed_image: bool = False
